packages/tpetra/core/src/Tpetra_Details_unpackCrsMatrixAndCombine_def.hpp

```diff
@@ -1533,11 +1533,10 @@ unpackAndCombineIntoCrsArrays (
     prefix << "permuteToLIDs.size() = " << permute_to_lids_d.size ()
     << "!= permute_from_lids_d.size() = " << permute_from_lids_d.size () << ".");
 
-  // Preseed TargetPids with -1 for local
   if (static_cast<size_t> (TargetPids.size ()) != TargetNumNonzeros) {
-    TargetPids.resize (TargetNumNonzeros);
+    Kokkos::resize(TargetPids,TargetNumNonzeros);
   }
-  TargetPids.assign (TargetNumNonzeros, -1);
+  Kokkos::parallel_for("initialize_TargetPids", TargetNumNonzeros, KOKKOS_LAMBDA (const int& i) { TargetPids(i) = -1; });
```

Comment:
```suggestion
  Kokkos::deep_copy(execution_space(), TargetPids, -1);
```

Reply:
Actually the parallel_for on the default exec space wasn't valid before, since we want Node=Serial to work on a build with Cuda enabled.

Reply:
addressed